### ISSUE TITLE
taskRoleArn Error

### ISSUE THREAD

**Issue by gdelgado** [fabfuel/ecs-deploy#14]
Hello,

Getting the error below when trying to deploy.

ecs deploy --profile my_profile --region us-east-1 my_cluster my_service --tag rel-0.1

Creating new task definition revision
Parameter validation failed:
Unknown parameter in input: "taskRoleArn", must be one of: family, containerDefinitions, volumes

**Comment by fabfuel** [comment]
Hi, 

what versions of `ecs-deploy` and `boto3` are you using?

TaskRoles are supported since version 1.1.0
https://github.com/fabfuel/ecs-deploy/releases

And you need at least boto3 >= 1.4.0

Best
Fabian

**Comment by gdelgado** [comment]
Hello @fabfuel.. I am using the boto3 version below grabbed from pip list. Also, there is no --version available in the tool, but I installed it from PIP.

boto3 (1.4.4)

Thanks..

**Comment by fabfuel** [comment]
Please have a look at this comment, looks like an issue with an old version of botocore
https://github.com/boto/boto3/issues/761#issuecomment-239233595

**Comment by gdelgado** [comment]
@fabfuel That did it..

Thanks.. One more question what is usually the recommended timeout setting.. Sometimes I see the tasks times out.. So I am guessing if I add the timeout flag it will more time for the API to respond.

**Comment by fabfuel** [comment]
Hi @gdelgado 

this time includes the full cycle of stopping all old and (re)starting all new containers. The overall deployment time depends on different things:
- the type of the application - I saw node.js containers wich take a very long time to get stopped. But nginx containers tend to stop immediately, etc.
- can old and new containers run in parallel (dynamic ports)?
- the deployment options and strategy (`Maximum percent` > 100)?
- the desired count of running tasks, compared to
- the number of ECS instances

If 600 sec works well for you, I don't see any reason why you should not use this value.

Best
Fabian

**Comment by joaoricardo000** [comment]
Just to add to this list:  
  
- If you have a web application service behind an ELB, add the time that it takes to deregister/register tasks for the blue/green deploy. This was taking ~5min in my case, for 2 tasks with max 200%.
